common/types.go

```diff
@@ -373,11 +373,37 @@ func (a Address) Value() (driver.Value, error) {
 	return a[:], nil
 }
 
-// Checks if an address is a valid account in our node's sharded address space
+// IsInChainScope checks if an address is a valid account in our node's sharded address space
 func (a Address) IsInChainScope() bool {
 	return NodeLocation.ContainsAddress(a)
 }
 
+// Location looks up the chain location which contains this address
+func (a Address) Location() *Location {
+	// Search zone chain address space
+	for r := 0; r < NumRegionsInPrime; r++ {
+		for z := 0; z < NumZonesInRegion; z++ {
+			l := Location{byte(r), byte(z)}
+			if l.ContainsAddress(a) {
+				return &l
+			}
+		}
+	}
+	// Search region chain address space
+	for r := 0; r < NumRegionsInPrime; r++ {
+		l := Location{byte(r)}
+		if l.ContainsAddress(a) {
+			return &l
+		}
+	}
+	// Is this address in Prime?
+	l := Location{}
```

Comment:
Since we expect zone TXs to be significantly more frequent than prime and region TXs (potentially by a few orders of magnitude), I decided to search the most likely group first, so that the average processing time is shorter. However; I'm happy to reorder the searches if you think I have it backwards.

Reply:
ok